### ISSUE TITLE
[Feat]: Frame drops

### ISSUE THREAD

**Issue by talhatsk** [apivideo/api.video-reactnative-live-stream#85]
### Version

2.0.0

### Environment that reproduces the issue

*

### Use case description

Hi, is there any method or callback that can tell us the frame drops in our stream?

**Comment by ThibaultBee** [comment]
Hi,

There isn't such APIs.
To understand the need, I have to understand:
- Why do you need it? (how do you plan to use it?)
- What is a frame? (video/audio/network packet)

**Comment by xoraingroup** [comment]
We would also like to have frame drops in some callback due to the reason, that some times when we stream to a specific RTMP server, the stream seems pixelated and stuttering. 

Having this option can enable us to adjust bitrate dynamically for high latency and poor internet connection users.

Frame drops are audio video frames. If its not possible, then a packet drops will suffice as well.

**Comment by talhatsk** [comment]
> Hi,
> 
> There isn't such APIs. To understand the need, I have to understand:
> 
> * Why do you need it? (how do you plan to use it?)
> * What is a frame? (video/audio/network packet)

yes i second @xoraingroup , the need is to change the bitrate when the users have high latency due to poor connection. to give them a seem less and fine streaming experience.

**Comment by ThibaultBee** [comment]
Ok, so you want to know if the connection is reliatable or not. That made more senses for RTMP than a callback with frame drops as RTMP is based on TCP.
I would rather add a small optional mechanism (maybe customizable with custom algorithm) to reduce bitrate when  sender internal buffer are too large.
But it is a complicated topic.
